### PR TITLE
refactor v2 api to be more compatible with v1

### PR DESCRIPTION
**Pull request by jeanseb6wind**
Parse data should use the builtin functions of libyang instead of reimplementing it from raw parsing.

### CODE REVIEW

**Comment by rjarry** [comment]
I'm not sure I like this. If libyang has changed their api, the bindings should follow the same nomenclature.

Is there a technical reason for this change or is it only a matter of preference?

**Comment by jeanseb6wind** [comment]
It would allow to not break current implementation using libyang-python (like sysrepo-python).

**Comment by rjarry** [comment]
I don't understand. The current implementation does **not** work with libyang2 anyway. It requires extensive changes: https://github.com/sysrepo/sysrepo-python/pull/28

Why not go the extra mile and adjust the calls to libyang-python API to use the new terminology?

**Comment by jeanseb6wind** [comment]
Yes I can, what do you prefer ?

**Comment by rjarry** [comment]
I prefer to use the new libyang2 api nomenclature.

File: libyang/context.py [hunk 1]
```diff
@@ -313,126 +313,85 @@ def parse_op_mem(
...
+                self.cdata, in_data, fmt, parse_flags, validate_flags, dnode
             )
-            lib.ly_in_free(data[0], 0)
-            if ret != lib.LY_SUCCESS:
-                raise self.error("failed to parse data tree")
-            return None
+        elif in_type == IOType.FD:
+            ret = lib.lyd_parse_data_fd(
+                self.cdata, in_data, fmt, parse_flags, validate_flags, dnode
+            )
```
Comment (rjarry) [inline libyang/context.py:350]: This is not related to "refactor v2 api to be more compatible with v1". Is it?

Comment (jeanseb6wind) [reply libyang/context.py:350]: indeed, I will create a new pr with only this update

**Comment by jeanseb6wind** [comment]
I close this pr.
